### PR TITLE
Feat/webpack 5

### LINKED ISSUE #1032: Improve developer experience and workflow
Currently loading the repo and running the basic initializing commands `yarn && yarn dev` doesn't work out of the box. I'd also like to improve the HMR of the extension so that it's smarter about how it's rebuilding and refreshing. I think we can get to a place much more in line with our other HMR projects.

### PR DESCRIPTION
**Pull request by aulneau**
> Try out this version of the Stacks Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/622528558), the [hosted version](https://pr-1044.app.stacks.engineering), or the [Stacks testnet demo app](https://pr-1044.testnet-demo.stacks.engineering)

I am _very_ excited about these changes!

This bumps our dep on webpack to v5, and refactors a lot of the config around it. This also updates the webpack configs to all live in `/app/wepback`, and implements a new dev server at `/app/webpack/dev-server.js`

The two tasks are now `yarn dev` and `yarn build`.

The big thing this gives us is `react-refresh` in the context of our extension!! I previously was under the impression that this was not possible, but thankfully it is. Now we get near instant updates with state persistence as we are working on the extension! see the video below :) 

This update also breaks out our original html file into 3 files, `full-width.html` `extension.html` and `popup.html` to give us better control over sizing when in different context.s

resolves #1032 

https://user-images.githubusercontent.com/11803153/109056943-be0b6a80-76a6-11eb-8980-7987fae0e574.mp4

### CODE REVIEW

**Comment by hstove** [comment]
Hey, the long story short here is that this seems great for the dev/prod build of the extension, but still many issues remaining from the upgrade for connect, the test app, and NODE_ENV=test stuff. I'm working through a lot of it at the moment.

**Comment by aulneau** [comment]
> Hey, the long story short here is that this seems great for the dev/prod build of the extension, but still many issues remaining from the upgrade for connect, the test app, and NODE_ENV=test stuff. I'm working through a lot of it at the moment.

Oh interesting, I'm happy to fix whatever is broken since it's my pr! Could you elaborate on what you mean? How is the webpack configuration for the extension related to connect?

**Comment by hstove** [comment]
Its actually really annoying and seems to be related to yarn and the monorepo. Even though the other packages depend on webpack 4, they're compiling with webpack 5. This is a root cause of a bunch of these checks failing.

```bash
# in packages/test-app
yarn webpack -v
# prints 4.x
yarn prod:web # or yarn prod:web, etc
# throws errors and finishes with "compiled with webpack 5.x"
```

I still could not figure out why this is happening. I basically gave up and have begun porting everything to webpack 5. I haven't made much progress - I have the test app ported to 5 for production builds, but don't have development build + hot reloading working yet.

**Comment by aulneau** [comment]
@hstove I see. I am happy to take a look if you like, since I already did the upgrade for the extension. Let me know and I can go through all the apps and ensure they are using a similar config/hmr etc. Or if you've already started, you can.

**Comment by hstove** [comment]
Thanks @aulneau ! I think it would be great if you were able to take a stab at it. I did want to try my hand at it to become more familiar with your changes and webpack 5 generally, but it's pretty nuanced and I imagine you've already dealt with many of the bugs that are popping up for me.

I pushed a WIP commit - it's not functional, just wanted to share what I have. Please feel free to completely squash it, it's kind of a grab bag of a few trial-and-error things.

**Comment by aulneau** [comment]
@hstove here we go -> https://github.com/blockstack/ux/commit/b4250e933960b74d5b0e53f0c02ce5ed4c2b8946 this fixes it. I more or less just moved over the entire webpack config from the ext and edited where needed. Would you be able to remove your last commit from this branch, and I can rebase into it?

**Comment by hstove** [comment]
Thanks! That's very helpful. I dropped my commit and added yours. Now I'll continue on making sure all the checks pass, which is easier.

**Comment by hstove** [comment]
The failed FF signing job is a misnomer (version conflict), so the only remaining CI thing is for updating Connect to webpack 5. Working on it!

File: packages/app/src/manifest.json [hunk 1]
```diff
@@ -8,26 +8,18 @@
...
-      "background.js"
-    ],
+    "scripts": ["background.js"],
     "persistent": true
   },
-  "web_accessible_resources": [
-    "inpage.js",
-    "index.html"
-  ],
+  "web_accessible_resources": ["inpage.js"],
```
Comment (aulneau) [inline packages/app/src/manifest.json:18]: @hstove this change seems to not work for me when I try to use the extension and auth with a new app. Have you tried this with a prod build?

**Review by hstove — APPROVED** [review]
Let's make sure CI passes and then good to merge!

**Comment by hstove** [comment]
~~CI passing - I re-added the extension reloader, then squashed all the commits into one. We'll double check that CI is passing and then anyone can click merge!~~

Scratch that, we are fixing some HMR stuff so I un-squashed this branch.

**Pull request merged**